llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -11509,6 +11510,19 @@ SDValue RISCVTargetLowering::lowerVECTOR_DEINTERLEAVE(SDValue Op,
     return DAG.getMergeValues(Res, DL);
   }
 
+  // TODO: Remove the e64 restriction once the fractional LMUL lowering
+  // is improved to always beat the vnsrl lowering below.
+  if (Subtarget.hasVendorXRivosVizip() && Factor == 2 &&
+      VecVT.getVectorElementType() == MVT::i64) {
```

Comment:
Can we do f64 too?

Reply:
Good idea, done.  And this is Factor 2 only, so the missing coverage isn't directly relevant.